### ISSUE TITLE
consider moving away from '&&'

### ISSUE THREAD

**Issue by maxogden** [meandavejustice/gh-pages-deploy#12]
for example, today i was debugging why my deploy step was failing. It was spawning this:

```
git checkout gh-pages && git reset --hard origin/master && npm run render && git add -A . && git commit -a -m 'gh-pages update' && git push origin gh-pages --force && git checkout master
```

what happened was the `git commit` step failed because i accidentally had `index.html` in my `.gitignore` and the commit exited with status 1 which caused the rest of the commands to fail

however, the `error` message that was returned from `child_process.exec` was `Command failed: git checkout gh-pages`. I think maybe node is doing something dumb here, but regardless it made me think the `git checkout` was failing when in fact it was the `commit` that was failing.

so I think a separate process should be spawned for each command, which would help with implementing e.g. #6 

also http://github.com/datproject/gasket would probably be useful in implementing this

**Comment by meandavejustice** [comment]
Will look into this over the weekend and let you know what I come up with

**Comment by ngoldman** [comment]
https://github.com/feross/standard uses a not bad way to spawn child processes internally: https://github.com/feross/standard/blob/master/index.js#L129-L137

**Comment by meandavejustice** [comment]
Thanks Nate,

I actually have a working version in a branch, I was just waiting for this
[issue](https://github.com/datproject/gasket/issues/18) in gasket It's been
a while tho so I should just merge my fix and create a new issue for when
that is updated.

Thanks for the reminder, I'll try and get this in tonight.

**Comment by meandavejustice** [comment]
fixed in  5a68f15e14d0f1da3ea0424611849b1e0ea20c34 

cc/ @maxogden @ngoldman
